prqlc/prqlc-ast/src/error.rs

```diff
@@ -112,15 +112,29 @@ impl std::fmt::Display for Errors {
 }
 
 pub trait WithErrorInfo: Sized {
+    fn span(&self) -> Option<&Span>;
+
     fn push_hint<S: Into<String>>(self, hint: S) -> Self;
 
     fn with_hints<S: Into<String>, I: IntoIterator<Item = S>>(self, hints: I) -> Self;
 
     fn with_span(self, span: Option<Span>) -> Self;
     fn with_code(self, code: &'static str) -> Self;
+
+    fn with_span_if_not_exists(self, span: Option<Span>) -> Self {
```

Comment:
No great stress, but wonder if this is something that should be done at the call site unless it's extremely prevalent (I only count one instance atm!), to avoid having a proliferation of methods here

Reply:
I had it at the call site and it was a mess of low level error handling in a high level function. But you're right, it can go into a new function there.